src/chain/mod.rs

```diff
@@ -3,58 +3,43 @@
 pub mod store;
 mod weight;
 use crate::blocks::Tipset;
+use crate::db::car::forest;
 use crate::ipld::stream_chain;
 use crate::utils::io::{AsyncWriterWithChecksum, Checksum};
 use anyhow::{Context, Result};
-use async_compression::futures::write::ZstdEncoder;
 use digest::Digest;
-use futures::future::Either;
-use futures::{io::BufWriter, AsyncWrite, AsyncWriteExt};
 use fvm_ipld_blockstore::Blockstore;
-use fvm_ipld_car::CarHeader;
-use std::sync::Arc;
+use tokio::io::{AsyncWrite, AsyncWriteExt, BufWriter};
 
 pub use self::{store::*, weight::*};
 
-pub async fn export<W, D>(
-    db: impl Blockstore + Send + Sync,
+pub async fn export<D: Digest>(
+    db: impl Blockstore,
     tipset: &Tipset,
     lookup_depth: ChainEpochDelta,
-    writer: W,
-    compressed: bool,
+    writer: impl AsyncWrite + Unpin,
```

Comment:
How about `writer: impl AsyncBufWrite + Unpin` to enforce using BufWriter here

Reply:
Do we want to enforce that?

Reply:
I see most of the `export` calls are passing File instead of BufWriter, should it be beneficial to use BufWriter given these exported snapshots are pretty large? if it is, should we enforce using BufWriter?

Reply:
`BufWriter` isn't automatically better, and it only improves performance when you have a lot of small writes. We do not have a lot of small writes.